### ISSUE TITLE
netcam pic_ready condition variable not used for rtsp cameras

### ISSUE THREAD

**Issue by jogu** [Motion-Project/motion#225]
We have  a pic_ready condition variable defined in netcam_context netcam.h:

```
pthread_cond_t pic_ready;   /* pthread condition structure used
                                for synchronisation between the
                                camera handler and the motion main
                                loop, showing new frame is ready */
```

However (apart from one call in netcam_init_jpeg, which isn't called at least in the rtsp camera case) nothing is waiting on this condition variable.

I suspect we should be waiting on this condition variable somewhere in the netcam case. But I'm not sure where. (Alternatively, If we shouldn't be waiting, we can remove a whole bunch of code in the rtsp case that signals it.)

**Comment by Mr-Dave** [comment]
I believe we should eliminate the notification signal.  The Motion loop should just pick up from the netcam loop whatever image is available when it wants one.  Slowing down the Motion loop waiting for a notification doesn't seem right.  Likewise determining which notifications to ignore when the netcam loop is running faster than Motion loopl wouldn't seem efficient either.

**Comment by jogu** [comment]
Yeah, sounds about right, thanks.

I may be over thinking this, but I think we probably should wait on the notification - it will probably almost always never wait (as there'll already be a frame ready) but the motion loop probably shouldn't ever analyse the same frame twice. [I'd guess this could happen on a camera running at say 2fps and when motion is running on a fast server.]

On the other side of the coin, I'm not sure if there's any work we can skip if the netcam is frames significantly faster than motion can analyse. I guess we might at least want to warn the user in that case, they could optimise overall CPU usage by reducing the frame rate.

**Comment by tosiara** [comment]
Do we want to keep this in the backlog?

**Comment by iskunk** [comment]
My branch has had quite a bit of churn around these condition variables, addressing not just the RTSP case but also the non-streaming case. (The latter has been rewritten to use sync barriers, which appears to be more robust---using both `pic_ready` and `cap_cond` together tends to lead to deadlocks. This was back when I was handling RTSP as a non-streaming camera, when the two threads ran in lockstep.)

I can get a PR in for this, though it would be a bit broader than what the title addresses.

**Comment by dorvan** [comment]
This cause a lock on captured stream in case of parallel captures of the same stream and the captured video it's like frozen, fragmented. Only on RTSP, not on RTMP .... this is one of the motivations why we want to capture RTMP streams instead of RTSP... #401

**Comment by Mr-Dave** [comment]
closed via #522
